pkg/cmd/attestation/verification/sigstore.go

```diff
@@ -165,7 +153,25 @@ func newCustomVerifier(trustedRootFilePath string) (*verify.SignedEntityVerifier
 		return nil, fmt.Errorf("failed to create trusted root from file %s: %v", trustedRootFilePath, err)
 	}
 
-	gv, err := verify.NewSignedEntityVerifier(trustedRoot, verify.WithSignedTimestamps(1))
+	verifierConfig := []verify.VerifierOption{}
+	verifierConfig = append(verifierConfig, verify.WithSignedCertificateTimestamps(1))
+	verifierConfig = append(verifierConfig, verify.WithObserverTimestamps(1))
+
+	// Infer verification options from contents of bundle and trusted root
+	bundleTimestamps, err := b.Timestamps()
+	if err != nil {
+		return nil, fmt.Errorf("unable to query bundle for timestamps")
+	}
+
+	if len(trustedRoot.TimestampingAuthorities()) > 0 && len(bundleTimestamps) > 0 {
+		verifierConfig = append(verifierConfig, verify.WithSignedTimestamps(1))
+	}
+
+	if len(trustedRoot.RekorLogs()) > 0 {
```

Comment:
Any specific reason here we don't check the bundle content as above for timestamps?

Reply:
I went back and forth on this - inferring verification policy from a custom trusted root and a bundle is more art than science! I reasoned that if your trusted root had Rekor logs, probably the bundle should have transparency logs?

This was inspired by, but is slightly different from, what we do in the sigstore-go conformance test: https://github.com/sigstore/sigstore-go/blob/3b9b707bb0ab79b30d0cfe4653294d9cd1a2cab8/cmd/conformance/main.go#L269

Reply:
My gut feeling is that this behaviour is preferable? That if there is a Rekor instance in the trust root, we require a transparency log entry. How about modifying line 166 above so that if there is a TSA in the trust root, we always require a signed timestamp in the bundle?

I get that this is a bit of guess work, but unless we accept a [ArtifactVerificationOptions](https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_verification.proto#L49) I think this is the best (most defensive) option we have.

Reply:
We do require at least one observer timestamp, regardless of if your bundle has a signed timestamp or not. I don't know why I thought of TSA as more optional than Rekor logs - I can make that change.